### PR TITLE
feat(server): Specify script flags with shebang

### PR DESCRIPTION
**Pull request by dranikpg**

### CODE REVIEW

File: src/server/main_service.cc [hunk 1]
```diff
@@ -991,27 +991,22 @@ void Service::Eval(CmdArgList args, ConnectionContext* cntx) {
...
-    return (*cntx)->SendError(result, facade::kScriptErrType);
-  }
+  auto res = server_family_.script_mgr()->Insert(body, interpreter);
+  if (!res)
+    return (*cntx)->SendError(res.error().Format(), facade::kScriptErrType);
 
-  if (add_result == Interpreter::ADD_OK) {
-    if (auto err = server_family_.script_mgr()->Insert(result, body); err)
-      return (*cntx)->SendError(err.Format(), facade::kScriptErrType);
-  }
```
Comment (dranikpg) [inline src/server/main_service.cc:1003]: All those changes are here, because we need to compile the script _without_ the shebang

Comment (romange) [reply src/server/main_service.cc:1003]: I do not understand. if it comes from a user why do you need to compile without it?
is it how Redis does it?

File: src/server/multi_test.cc [hunk 2]
```diff
@@ -657,22 +657,22 @@ TEST_F(MultiTest, ScriptConfig) {
...
 )";
 
-  // Check eval finds script pragmas.
+  // Check eval finds script flags.
   Run({"set", "random-key", "works"});
   EXPECT_EQ(Run({"eval", s1, "0"}), "works");
 
   const char* s2 = R"(
-  -- pragma: this-is-an-error
+  #!lua flags=this-is-an-error
```
Comment (romange) [inline src/server/multi_test.cc:671]: btw, does it mean it's an invalid lua code?

File: src/server/script_mgr.cc [hunk 3]
```diff
@@ -185,38 +174,49 @@ void ScriptMgr::LatencyCmd(ConnectionContext* cntx) const {
   }
 }
 
-// Search for pragmas in script body
-io::Result<optional<ScriptMgr::ScriptParams>, GenericError> DeduceParams(string_view body) {
-  static const regex kRegex{"^\\s*?--\\s*?pragma\\s*?:(.*)"};
+// Search for flags shebang if present and truncate shebang.
```
Comment (romange) [inline src/server/script_mgr.cc:177]: should not it be the first line in the script or something?

File: src/server/multi_test.cc (hunk 2 above)
Comment (dranikpg) [reply src/server/multi_test.cc:671]: Must be, it fails to compile. Also note that we insert it into a function when we store the script

File: src/server/script_mgr.cc (hunk 3 above)
Comment (dranikpg) [reply src/server/script_mgr.cc:177]: Yes, I'll rename it to check for flags shebang. The regex checks for start

File: src/server/main_service.cc (hunk 1 above)
Comment (dranikpg) [reply src/server/main_service.cc:1003]: Yes, its invalid lua code

**Comment by dranikpg** [comment]
Should we mark existing redis flags (no-writes,allow-stale) as invalid? Or silently ignore them? Btw `no-writes` is useful

**Review by romange — APPROVED** [review]

**Pull request merged**